### PR TITLE
Feature/90 filter recipes

### PR DESCRIPTION
**Pull request by 123kevyd**
Note: I want to go ahead and stealth merge xians branch, so I just merged his code into this branch. I would wait until his branch is merged before merging mine

### CODE REVIEW

File: components/recipe_table_comp.js [hunk 1]
```diff
@@ -33,6 +51,67 @@ class RecipeTable extends Component {
...
+            return toRemoveTitle
+        }
+
+        shownRecipes.forEach((recipe) => {
+            let toRemoveTitle = null
+            recipe.key = recipe.id
+            recipe.rating = this.getAverageVal(recipe.reviews, 'stars')
+            recipe.difficulty = this.getAverageVal(recipe.reviews, 'difficulty')
+
+            toRemoveTitle = checkRecipeArray(useIngredientFilter, this.props.myIngredients, recipe, "ingredients", (elem2, elem) => { return elem.title === elem2.name })
```
[the remark this answers is not in the document]
Comment (StoicDeveloper) [reply components/recipe_table_comp.js:99]: also, don't abbreviate compare to comp, it reduces readability and saves 2 characters.

File: components/recipe_table_comp_HOC.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { userStore } from "/store/user_store"
+import RecipeTable from './recipe_table_comp';
+
+export default function RecipeTableHOC(props) {
```
[the remark this answers is not in the document]
Comment (123kevyd) [reply components/recipe_table_comp_HOC.js:5]: HOC apparently for high order component
Here is the page on them from the react docs:
https://reactjs.org/docs/higher-order-components.html

File: components/recipe_table_comp.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (123kevyd) [reply components/recipe_table_comp.js:99]: I was able to remove the .key to .id one, however, the two functions are not just naming issues. The average rating and difficulty are not in the original recipe, and the easiest way to get them to be displayed in the table is by adding them as a field here. This also allows the values to be updated as reviews come in.

File: components/recipe_table_comp.js [hunk 3]
```diff
@@ -33,6 +51,67 @@ class RecipeTable extends Component {
...
+
+        let shownRecipes = this.formatRecipes(this.props.recipes, newFilters.useIngredientFilter, newFilters.useEquipmentFilter, newFilters.useRestrictionFilter, newFilters.useRecipesFilter);
+        this.setState({shownRecipes: shownRecipes})
+	}
+
+    formatRecipes(recipes, useIngredientFilter, useEquipmentFilter, useRestrictionFilter, useRecipesFilter) {
+        let toRemove = []
+        let shownRecipes = [...recipes]
+
+        function checkRecipeArray(filterBool, filterList, recipe, fieldName, compFunc) {
```
[the remark this answers is not in the document]
Comment (123kevyd) [reply components/recipe_table_comp.js:69]: I have fixed the first two suggestions but will wait to see if that becomes a performance issue before fixing the last one.

File: components/recipe_table_comp.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (StoicDeveloper) [reply components/recipe_table_comp.js]: The minimum size of the code window in these conversations appears to be 4 lines; I had no objection to creating the rating and recipe fields.

**Pull request merged**